File: src/dataset/python/rotational-cipher_exercism-fruit-in/claude/ClaudeSonnet4_rotational_cipher_v1.py
def rotate(text, key):
    key = key % 26
    result = []
    
    for char in text:
        if 'A' <= char <= 'Z':
            result.append(chr((ord(char) - 65 + key) % 26 + 65))
        elif 'a' <= char <= 'z':
            result.append(chr((ord(char) - 97 + key) % 26 + 97))
        else:
            result.append(char)
    
    return ''.join(result)

File: src/dataset/typescript/rotational-cipher_exercism-wackerow/claude/ClaudeSonnet4_rotational-cipher_v1.ts
export default class RotationalCipher {
  private static readonly _abc = 'abcdefghijklmnopqrstuvwxyz'
  private static readonly _ABC = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ'

  public static rotate(message: string, key: number): string {
    const result: string[] = []
    const normalizedKey = ((key % 26) + 26) % 26
    
    for (let i = 0; i < message.length; i++) {
      const char = message[i]
      const code = char.charCodeAt(0)
      
      if (code >= 97 && code <= 122) { // a-z
        result.push(String.fromCharCode(((code - 97 + normalizedKey) % 26) + 97))
      } else if (code >= 65 && code <= 90) { // A-Z
        result.push(String.fromCharCode(((code - 65 + normalizedKey) % 26) + 65))
      } else {
        result.push(char)
      }
    }
    
    return result.join('')
  }
}